health-services/project/src/main/java/org/egov/project/repository/LocationCaptureRepository.java

```diff
@@ -0,0 +1,133 @@
+            query = query + " WHERE ul.tenantId=:tenantId ";
+        } else {
+            query = query + " AND ul.tenantId=:tenantId ";
+        }
+
+        if (urlParams.getLastChangedSince() != null) {
+            query = query + " AND ul.lastModifiedTime>=:lastModifiedTime ";
+        }
+        paramsMap.put("tenantId", urlParams.getTenantId());
+        paramsMap.put("lastModifiedTime", urlParams.getLastChangedSince());
+
+        try {
+            log.debug("Executing query to fetch total count");
+            Long totalCount = CommonUtils.constructTotalCountCTEAndReturnResult(query, paramsMap, this.namedParameterJdbcTemplate);
+
+            query = query + " ORDER BY ul.id ASC LIMIT :limit OFFSET :offset";
+            paramsMap.put("limit", urlParams.getLimit());
+            paramsMap.put("offset", urlParams.getOffset());
+
+            log.debug("Executing query to fetch user locations: {}", query);
+            List<UserAction> locationCaptureList = this.namedParameterJdbcTemplate.query(query, paramsMap, this.rowMapper);
+
+            log.info("Successfully fetched user locations: {}", locationCaptureList.size());
+            return SearchResponse.<UserAction>builder().response(locationCaptureList).totalCount(totalCount).build();
+        } catch (Exception e) {
+            log.error("Failed to execute query for finding user locations", e);
+            return SearchResponse.<UserAction>builder().response(Collections.emptyList()).totalCount(0L).build();
+        }
+    }
+
+    /**
+     * Finds user locations by their IDs, first checking the cache before querying the database.
+     *
+     * @param ids         The list of IDs to search for.
+     * @param columnName  The name of the column to search by.
+     * @return A SearchResponse containing the list of user locations found.
+     */
+    public SearchResponse<UserAction> findById(List<String> ids, String columnName) {
+        log.info("Executing findById with ids: {} and columnName: {}", ids, columnName);
+
+        List<UserAction> objFound = findInCache(ids);
+
+        if (!objFound.isEmpty()) {
+            Method idMethod = getIdMethod(objFound, columnName);
+            ids.removeAll(objFound.stream()
+                    .map(obj -> (String) ReflectionUtils.invokeMethod(idMethod, obj))
+                    .collect(Collectors.toList()));
+
+            if (ids.isEmpty()) {
+                log.info("All requested user locations found in cache");
+                return SearchResponse.<UserAction>builder().response(objFound).build();
+            }
+        }
+
+        String query = String.format("SELECT id, clientreferenceid, tenantid, projectid, latitude, longitude, locationaccuracy, boundarycode, action, createdby, createdtime, lastmodifiedby, lastmodifiedtime, clientcreatedtime, clientlastmodifiedtime, clientcreatedby, clientlastmodifiedby, additionaldetails FROM user_location ul WHERE ul.%s IN (:ids)", columnName);
```

Comment:
same query written twice

Reply:
will store in a variable and utilise using that.

Reply:
done